### DISCUSSION TITLE
Can any app that uses LXMF communicate with any other app that uses LXMF?

### DISCUSSION THREAD

**Discussion by trosel** [markqvist/Reticulum#53] [Help & Questions] [answered]
I assume that users on Nomad and users on Sideband can both message each other back and forth?

The Nomad app is called the "Nomad *Network*" on your website, so that is what makes me think maybe the app is its own self-contained network, separate from the Sideband app.

In my head, the whole ecosystem is Reticulum *Network*, with LXMF being the preferred format for message-sending, and any app that speaks LXMF can talk to each other.

**Comment by markqvist** [comment, marked as answer]
The short answer is: Yes, any LXMF app can communicate with any other LXMF client. The format is completely flexible though, so one client might choose to implement features that others don't necessarily support.

Sideband and Nomad Network is completely interoperable. You can message one from the other and vice versa.

Nomad Network also includes functions for hosting sites, pages and files, and for providing infrastructure (propagation nodes) for LXMF users, which is the reason behind the naming.

There is definitely room for improvements in all of the terminology, but I hope this clears it up a little.

**Comment by trosel** [comment]
Thank you! I plan to try to make a messaging app as well with LXMF. Just need to find a good GUI library that will mesh well with the Python SDK. Maybe Flutter can connect to it. 

regarding the branding, I think it would be smart to consider this line of thinking https://snikket.org/blog/products-vs-protocols/

**Comment by markqvist** [comment]
Very cool! I will gladly help you with anything related to the backend and the LXMF API. You can use the Sideband source code as an example of how to do all the basic backend stuff needed to communicate over LXMF. The nomadnet code is a little more... unruly... at the moment ;) I will have the full API documentation for LXMF written soon, but you are welcome to ask any questions here in the meantime.

Thanks for the link, I'll read it when I get a moment. From the looks of it, I totally agree, and really, what I am hoping for is people like you to step in and create the well-branded, general-purpose clients. I only have so much time, and I really want to prioritise it where it is best spend, which I think is in the core protocol, utilities, documentation, learning resources and a few specific apps.
